### PR TITLE
Frontend tool cleanup

### PR DESCRIPTION
**Pull request by slavapestov**
Fine-grained dependencies are enabled by default in the 5.3 branch, and we’re not re-branching again, so let’s rip out the old dependency graph implementation on master.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Review by CodaFi — APPROVED** [review]
GO GO GO

:shipit:

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: include/swift/Driver/Compilation.h [hunk 1]
```diff
@@ -275,11 +275,7 @@ class Compilation {
   const bool OnlyOneDependencyFile;
 
 private:
-  /// Scaffolding to permit experimentation with finer-grained dependencies and
-  /// faster rebuilds.
-  const bool EnableFineGrainedDependencies;
-
-  /// Is the parser recording token hashes for each type body?
+  //// Is the parser recording token hashes for each type body?
```
Comment (JaSpa) [inline include/swift/Driver/Compilation.h:278]: You have one slash too many here :)

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**